taichi/runtime/llvm/runtime.cpp

```diff
@@ -1276,13 +1276,13 @@ struct printf_helper {
 
 template <typename... Args>
 void taichi_printf(LLVMRuntime *runtime, const char *format, Args &&... args) {
-#if ARCH_cuda
-  printf_helper helper;
-  helper.push_back(std::forward<Args>(args)...);
-  vprintf((Ptr)format, helper.ptr());
-#else
+//#if ARCH_cuda
+  //printf_helper helper;
+  //helper.push_back(std::forward<Args>(args)...);
+  //vprintf((Ptr)format, helper.ptr());
+//#else
   runtime->host_printf(format, args...);
```

Comment:
Actually we cannot use `host_printf` on CUDA. `vprintf` is the only way to print on a CUDA thread.

Reply:
The real problem is `clang` secretly includes `<cstdio>`...

Reply:
Right. Thanks for trying this - I learn how tricky it is to solve this within C++. Let me try to fix this issue using LLVM. I'll open a separate PR for this so that you can review it.